pkg/vsphere/disk/disk_manager.go

```diff
@@ -281,7 +288,9 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 	machineSpec := types.VirtualMachineConfigSpec{}
 	machineSpec.DeviceChange = append(machineSpec.DeviceChange, changeSpec...)
 
-	m.reconfig.Lock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
```

Comment:
That's racy - as best I can tell there's no lock that would prevent a parallel call detaching before we call setAttached.